lib/model/query/datasets.js

```diff
@@ -456,9 +456,14 @@ LEFT JOIN (
   JOIN entity_def_sources es ON es.id = ed."sourceId"
   JOIN entities e ON e.id =  ed."entityId" AND e."datasetId" = ${datasetId}
 ) ON es."submissionDefId" = sd.id
+LEFT JOIN (
+  SELECT DISTINCT ON ((details->'submissionId')::INTEGER) * FROM audits
```

Comment:
Should we use `submissionDefId` instead of `submissionId`? If the submission had an error, then an edit, and no error specifically associated with the edit, it should still be processed I think.

Reply:
I thought about this, but the way this would happen is
- submission
- approval
- error
- edit (new sub def)
- DON'T approve yet
- change dataset flag

... and that seems so specialized to have the auto processing happen then. 

Someone could approve earlier or could do the edit after changing the dataset flag and it should work.

It was easier for me to think about how if the submission had ever caused an error with an entity, it wouldn't get caught up in this reprocessing scenario.

Reply:
Changing the code so it _will_ process any new version of a submission that hasn't specifically generated an error.